ntp-proto/src/peer.rs

```diff
@@ -51,44 +50,43 @@ pub struct Peer {
 /// If the register contains any nonzero bits, the server is considered reachable;
 /// otherwise, it is unreachable.
 #[derive(Debug, Default, Clone, Copy)]
-pub(crate) struct Reach(u8);
+struct Reach(u8);
 
 impl Reach {
-    #[allow(dead_code)]
     fn is_reachable(&self) -> bool {
         self.0 != 0
     }
 
     /// We have just received a packet, so the peer is definitely reachable
-    #[allow(dead_code)]
     fn received_packet(&mut self) {
         self.0 |= 1;
     }
 
     /// A packet received some number of poll intervals ago is decreasingly relevant for
     /// determining that a peer is still reachable. We discount the packets received so far.
-    #[allow(dead_code)]
     fn poll(&mut self) {
         self.0 <<= 1
     }
 }
 
-#[allow(dead_code)]
-pub enum MsgForSystem {
-    NoUpdate,
-    PeerUpdated(PeerUpdated),
+pub enum PeerError {
+    /// The association mode is not one that this peer supports
+    InvalidMode,
+    /// The send time on the received packet is not the time we sent it at
+    InvalidPacketTime,
+    /// Peer is in an invalid state
+    Kiss,
```

Comment:
Not all of these represent error conditions perse, so calling this an error is a bit weird.